File: packages/ts-sdk/src/treedx/adapters/snapshots.ts
import { jsonRequest, segment, type TreeDxAdapterContext } from './common.js';

export class SnapshotsAdapter {
  constructor(private readonly context: TreeDxAdapterContext) {}
  build(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/snapshots/build`, input); }
  get(repoId: string, snapshotId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/snapshots/${segment(snapshotId)}`); }
}

File: packages/elixir-sdk/lib/treedx/adapters/storage/snapshots.ex
defmodule TreeDxSdk.Snapshots do
  @moduledoc false
  alias TreeDxSdk.Adapters.Common

  def build(client, repo_id, body) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/snapshots/build",
      body,
      %{}
    )
  end

  def get(client, repo_id, snapshot_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/snapshots/" <> Common.segment(snapshot_id),
      nil,
      %{}
    )
  end
end
